modules/components/stories/mappingToTreeData.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+const MOCK_MAPPING = {
+  boolean: {
+    type: 'boolean',
+  },
+  children: {
+    properties: {
+      key: {
+        type: 'text',
+        fields: {
+          keyword: {
+            type: 'keyword',
+            ignore_above: 256,
+          },
+        },
+      },
+    },
+  },
+  float: {
+    type: 'float',
+  },
+  int: {
+    type: 'long',
+  },
+  stringarray: {
+    type: 'text',
+    fields: {
+      keyword: {
+        type: 'keyword',
+        ignore_above: 256,
+      },
+    },
+  },
+  text: {
+    type: 'text',
+    fields: {
+      keyword: {
+        type: 'keyword',
+        ignore_above: 256,
+      },
+    },
+  },
+};
+const elasticMappingToDisplayTreeData = elasticMapping => {
+  const mappingKeys = Object.keys(elasticMapping);
+  return mappingKeys.map(key => {
+    const fieldProps = elasticMapping[key];
+    return {
+      title: key,
+      id: key,
+      [fieldProps.properties && 'children']: fieldProps.properties
```

Comment:
Double check on fieldProps.properties, this should probably just be children always. 
if you need children to not exist if `fieldProps.properties` is falsy, you can conditionally spread. 
```
...(fieldProps.properties ? {children: elasticMappingToDisplayTreeData(fieldProps.properties) }) : {}
```

Reply:
I see, yes that makes more sense